src/anndata/typing.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Union
+
+import numpy as np
+import pandas as pd
+from numpy import ma
+from scipy import sparse
+
+from . import abc
+from ._core.anndata import AnnData
+from .compat import (
+    AwkArray,
+    CupyArray,
+    CupySparseMatrix,
+    DaskArray,
+    H5Array,
+    SpArray,
+    ZappyArray,
+    ZarrArray,
+)
+
+if TYPE_CHECKING:
+    from typing import TypeAlias
+
+
+__all__ = ["RWAble", "AxisStorable"]
+
+
+ArrayDataStructureType: TypeAlias = Union[
+    np.ndarray,
+    ma.MaskedArray,
+    sparse.csr_matrix,
+    sparse.csc_matrix,
+    SpArray,
+    AwkArray,
+    H5Array,
+    ZarrArray,
+    ZappyArray,
+    abc.CSRDataset,
+    abc.CSCDataset,
+    DaskArray,
+    CupyArray,
+    CupySparseMatrix,
+]
```

Comment:
What's supposed to be the difference between this file and `_types` if this isn't public?

Reply:
Everything in here is public, just not all names.

There are a few internal APIs that use subsets of the public types, and these are included in here too.

I’d have put these into `_types` if Python was able to deal with this kind of circularity.

So I guess the logic is “all public types and the types they depend on?

Reply:
Sounds good!